Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/PartitionSceneEntitiesSystem.cs

```diff
@@ -95,39 +99,33 @@ protected override void Update(float t)
         [None(typeof(PartitionComponent))]
         private void PartitionNewEntity(in Entity entity, ref SceneDefinitionComponent definition)
         {
+            PartitionComponent partitionComponent = partitionComponentPool.Get();
+            World.Add(entity, partitionComponent);
+
             if (definition.IsPortableExperience)
             {
-                PartitionComponent partitionComponent = partitionComponentPool.Get();
                 partitionComponent.OutOfRange = false;
                 partitionComponent.Bucket = 0;
                 partitionComponent.IsBehind = false;
                 partitionComponent.RawSqrDistance = 1;
                 partitionComponent.IsDirty = true;
-                World.Add(entity, partitionComponent);
                 return;
             }
```

Comment:
I think if we do the `World.Add()` of a component BEFORE manipulating it later (like your new scenario for `definition.IsPortableExperience` cases), we risk having some obscure (Arch)ECS problems, take a look at this recommendation in the Wiki: https://github.com/decentraland/unity-explorer/wiki/How-To#how-to-manipulate-components

I'm not 100% sure your case falls into that risk, but it looks like it may be the case, maybe @mikhail-dcl can confirm

Reply:
Cool now it looks good to me =)

>why doesn't the query iterator not break on any archetype change at all during its run
I have no idea, there is some black magic on Arch that I haven't been able to decipher yet (nor had time to dive into)